### DISCUSSION TITLE
Upload Folder

### DISCUSSION THREAD

**Discussion by AcrNischal** [community/community#112072] [New to GitHub] [answered]
I am not able to upload my folder to github, how can I?

**Comment by ParasRupani** [comment, marked as answer]
Hi Nishchal, here is a link where your question is already answered: [How to upload an entire folder?](https://github.com/orgs/community/discussions/21962)

What I would do in your shoes is:
Go to the repository/folder in which you want to upload your folder and then:
Choose- "Add file" - Upload files
Drop the folder you wish to upload by dragging it into the box.

Else you can also follow this from the terminal:
Make sure your working directory is the directory where you have your folder to upload then:
git remote set-url origin <new_repository_url>
git add <folder>
git commit -m "Your commit message"
git fetch origin
git merge origin/main
git push origin main

Let me know if this works.

**Comment by probablynotnmp** [comment]
Have you tried just dragging and dropping the whole folder from your desktop into the repo after you clicked on "upload files"

**Comment by Ananya46Nigam** [comment]
To streamline the process of uploading folders to your GitHub repository, you have a couple of options. One way is to manually create folders within your GitHub repository by dragging and dropping them from your local directory. However, this method can be quite time-consuming.

Alternatively, you can leverage Git to automate this process. Here's how you can do it:

1. Ensure Git is installed on your system. You can verify this by typing "git" in your command prompt.

2. To upload a folder:
   - Create a repository on GitHub if you haven't already done so.
   - Right-click on the folder you wish to upload to your repository.
   - Select "Git Bash here" from the dropdown menu.
   - In the Git Bash terminal, execute the following commands:

     git init
     git add .
     git commit -m "Specify a message indicating the changes being made"
     git branch -M main  # Ensure you're working with the correct branch (main or master)
     git remote add origin <HTTPS link of your repository, obtainable by clicking on Code>
     git push -u origin main


3. You can verify whether all your folder files have been successfully uploaded by using "git status". If any files are missing, you can add them individually using "git add <filename>".

**Comment by jfullstackdev** [comment]
both GitHub Web and GitHub Desktop have the drag-n-drop feature of folders being uploaded

**Reply by Ananya46Nigam** [reply]
That does not upload folder with files which are also inside folders
